contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1181 @@
+     * underlying rooster pool. Print the required amount of corresponding OETHp. After the rebalancing is
+     * done burn any potentially remaining OETHp tokens still on the strategy contract.
+     *
+     * This function has a slightly different behaviour depending on the status of the underlying Rooster
+     * pool. The function consists of the following 3 steps:
+     * 1. withdrawLiquidityOption -> this is a configurable option where either only part of the liquidity
+     *                               necessary for the swap is removed, or all of it. This way the rebalance
+     *                               is able to optimize for volume, for efficiency or anything in between
+     * 2. swapToDesiredPosition   -> move active trading price in the pool to be able to deposit WETH & OETHp
+     *                               tokens with the desired pre-configured ratios
+     * 3. addLiquidity            -> add liquidity into the pool respecting ratio split configuration
+     *
+     *
+     * Exact _amountToSwap, _swapWeth & _minTokenReceived parameters shall be determined by simulating the
+     * transaction off-chain. The strategy checks that after the swap the share of the tokens is in the
+     * expected ranges.
+     *
+     * @param _amountToSwap The amount of the token to swap
+     * @param _swapWeth Swap using WETH when true, use OETHb when false
+     * @param _minTokenReceived Slippage check -> minimum amount of token expected in return
+     * @param _liquidityToRemovePct Percentage of liquidity to remove -> the percentage amount of liquidity to
+     *        remove before performing the swap. 1e18 denominated
+     */
+    function rebalance(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived,
+        uint256 _liquidityToRemovePct
+    ) external nonReentrant onlyGovernorOrStrategist {
+        _rebalance(
+            _amountToSwap,
+            _swapWeth,
+            _minTokenReceived,
+            _liquidityToRemovePct
+        );
+    }
+
+    // slither-disable-start reentrancy-no-eth
+    function _rebalance(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived,
+        uint256 _liquidityToRemovePct
+    ) internal {
+        // Remove the required amount of liquidity
+        if (_liquidityToRemovePct > 0) {
+            _removeLiquidity(_liquidityToRemovePct);
+        }
+
+        // In case liquidity has been removed and there is still not enough WETH owned by the
+        // strategy contract remove additional required amount of WETH.
+        if (_swapWeth && _amountToSwap > 0) {
+            _ensureWETHBalance(_amountToSwap);
+        }
+
+        // in some cases (e.g. deposits) we will just want to add liquidity and not
+        // issue a swap to move the active trading position within the pool
+        if (_amountToSwap > 0) {
+            _swapToDesiredPosition(_amountToSwap, _swapWeth, _minTokenReceived);
+        }
```

Comment:
What do you think of combining this two sections in a single one? Something like this:
```solidity
// in some cases (e.g. deposits) we will just want to add liquidity and not
// issue a swap to move the active trading position within the pool
if (_amountToSwap > 0) {
    // In case liquidity has been removed and there is still not enough WETH owned by the
    // strategy contract remove additional required amount of WETH.
    if (_swapWeth) _ensureWETHBalance(_amountToSwap);
    
    _swapToDesiredPosition(_amountToSwap, _swapWeth, _minTokenReceived);
}
```
It reduces one check, but more important for me, it groups actions.

Reply:
I like the suggestion, thanks: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/97af511f7a980410133e66a11485565d5ebb0643